src/history/hs_rec.c

```diff
@@ -1057,17 +1031,28 @@ __hs_delete_key_from_pos(
         /*
          * Append a globally visible tombstone to the update list. This will effectively make the
          * value invisible and the key itself will eventually get removed during reconciliation.
+         *
+         * If anything fails after this point we need to panic as it will leave our history store in
+         * an unexpected state with duplicate entries.
          */
-        WT_ERR(__wt_upd_alloc_tombstone(session, &upd, NULL));
+        if ((ret = __wt_upd_alloc_tombstone(session, &upd, NULL)) != 0)
```

Comment:
Just an idea: since the initialization of this `upd` is pretty simple (its timestamp and txnid info just gets 0'd out), can we allocate it before the `__hs_insert_record_with_btree_int` call? It'd mean we can remove one panic here.

Reply:
7a4faea